TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -49,19 +55,85 @@ private SelectClause CreateSelectClause(CodeElementsParser.Sql_selectClauseConte
             SyntaxProperty<SelectionModifier> selectionModifier = null;
             if (context.SQL_ALL() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All, ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All,
+                    ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
             }
             else if (context.SQL_DISTINCT() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct, ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct,
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
             }
+
             if (context.star() != null)
             {
                 return new SelectClause(selectionModifier, new StarSelection());
             }
 
+            if (context.selections() != null)
+            {
+                var tableOrViewAllColumnsSelections = context.selections().selection().Select(CreateSelection)
+                    .Where(selection => selection != null).ToList();
+
+                return new SelectClause(selectionModifier, tableOrViewAllColumnsSelections);
+            }
+
+            return null;
+        }
+
+        private TableOrViewAllColumnsSelection CreateSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.tableOrViewAllColumnsSelection() != null)
+            {
+                var tableOrViewOrCorrelationName =
+                    context.tableOrViewAllColumnsSelection().tableOrViewOrCorrelationName();
```

Comment:
Could you test what happens on a syntax like `SELECT .*` ? Does ANTLR produce an error and we don't even enter this method or `context.tableOrViewAllColumnsSelection().tableOrViewOrCorrelationName()`is null and then we have an exception ?